dotnet/src/SemanticKernel.Abstractions/Data/TextSearch/TextSearchResultPropertyReader.cs

```diff
@@ -0,0 +1,92 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Reflection;
+
+namespace Microsoft.SemanticKernel.Data.TextSearch;
```

Comment:
Would this make more sense in Core considering that it's only used from there?
Also should this be in the Data namespace with the other classes?